github/teams.go

```diff
@@ -252,11 +279,24 @@ func (s *TeamsService) DeleteTeam(ctx context.Context, team int64) (*Response, e
 	return s.client.Do(ctx, req, nil)
 }
 
-// ListChildTeams lists child teams for a team.
+// DeleteTeamByName deletes a team reference by Name.
+//
+// GitHub API docs: https://developer.github.com/v3/teams/#delete-team
+func (s *TeamsService) DeleteTeamByName(ctx context.Context, org, slug string) (*Response, error) {
```

Comment:
For consistency, please rename this function to `DeleteTeamBySlug` and change comment to day "reference by slug."